lib/ansible/modules/system/lvol_thin.py

```diff
@@ -0,0 +1,755 @@
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+author:
+    - "Pavol Ipoth <pavol.ipoth@gmail.com>"
+module: lvol_thin
+version_added: 2.4
+short_description: Configure Thin LVM logical volumes
+description:
+  - This module creates, removes or resizes, converts thin logical volumes.
+options:
+  vg:
+    description:
+    - The volume group this logical volume is part of.
+    required: true
+  lv:
+    description:
+    - The name of the logical volume.
+    required: true
+  size:
+    description:
+    - The size of the logical volume, by
+      default in megabytes or optionally with one of [bBsSkKmMgGtTpPeE] units; or
+      as a percentage of [VG|PVS|FREE]; +size is not implemented, as it is by
+      design not indempodent
+      Float values must begin with a digit.
+  state:
+    choices: [ "present", "absent" ]
+    default: present
+    description:
+    - Control if the logical volume exists. If C(present) the C(size) option
+      is required.
+    required: false
+  force:
+    choices: [ "yes", "no" ]
+    default: "no"
+    description:
+    - Shrink/extend/remove/conversion operations of volumes requires this switch. Ensures that
+      that filesystems get never corrupted/destroyed by mistake.
+    - this flag always means that data on lvol will be destroyed
+    required: false
+  opts:
+    description:
+    - Free-form options to be passed to the lvcreate command
+  pvs:
+    description:
+    - list of physical volumes /dev/sda /dev/sdc
+  type:
+    choices: ["thin", "thin-pool", "normal"]
+    default: normal
+    description:
+    - Type of logical volume, thin is thind lvol, thin-pool is thin pool
+      supplying thind lvol, normal is lvol without any type. To create thin
+      lvol, thin pool must already exist
+  pool:
+    description:
+    - This value is used when type is thin, specifies which thin pool will
+      be caching data for thin lvol. Must be in format vg/lv. Thin pool lvol
+      and thin lvol must be from same vg (this is restriction of lvm2).
+notes:
+  - Filesystems on top of the volume are not resized.
+'''
+
+EXAMPLES = '''
+# Create a logical volume of 512m.
+- lvol: vg=firefly lv=test size=512m
```

Comment:
Don't use this on-liner style. Use proper YAML formatting. Also use the comment like a name of the task (no dot at the end):

```
- name: Create a logical volume of 512m
  lvol:
    vg: firefly
    lv: test
    size: 512m
```

The same applies also for all examples bellow.